ipaserver/install/ipa_backup.py

```diff
@@ -619,33 +652,45 @@ def finalize_backup(self, data_only=False, encrypt=False, keyring=None):
         '''
 
         if data_only:
-            backup_dir = os.path.join(paths.IPA_BACKUP_DIR, time.strftime('ipa-data-%Y-%m-%d-%H-%M-%S'))
+            backup_dir = os.path.join(
+                paths.IPA_BACKUP_DIR,
+                time.strftime('ipa-data-%Y-%m-%d-%H-%M-%S')
+            )
             filename = os.path.join(backup_dir, "ipa-data.tar")
         else:
-            backup_dir = os.path.join(paths.IPA_BACKUP_DIR, time.strftime('ipa-full-%Y-%m-%d-%H-%M-%S'))
+            backup_dir = os.path.join(
+                paths.IPA_BACKUP_DIR,
+                time.strftime('ipa-full-%Y-%m-%d-%H-%M-%S')
+            )
             filename = os.path.join(backup_dir, "ipa-full.tar")
 
-        os.mkdir(backup_dir)
-        os.chmod(backup_dir, 0o700)
-
-        os.chdir(self.dir)
-        args = ['tar',
-                '--xattrs',
-                '--selinux',
-                '-czf',
-                filename,
-                '.'
-               ]
-        result = run(args, raiseonerr=False)
-        if result.returncode != 0:
+        try:
+            os.mkdir(backup_dir)
+            os.chmod(backup_dir, 0o700)
+        except Exception as e:
```

Comment:
Similar as here: https://github.com/freeipa/freeipa/pull/3112#pullrequestreview-233924039

Reply:
os.mkdir would fail with OSError I think.

Reply:
Of course @fcami I only meant to specify Exception type but thanks for correction :)

Reply:
In fact that's what I'm concerned about @Tiboris - Python 2.7 would behave significantly differently if I remember correctly (IOError vs OSError).
As the try statement only protects OS-specific calls I'm not sure what would be gained by catching specific exceptions there. Also ipa_backup.py already behaves that way everywhere (see exception handling after LDAP connections).

Reply:
Resolved, see above. Thanks!